src/Raven.Server/Documents/Indexes/Static/TimeSeries/DynamicTimeSeriesSegment.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Raven.Server.Documents.TimeSeries;
+using Raven.Server.Smuggler.Documents;
+using Raven.Server.Utils;
+using Sparrow.Extensions;
+
+namespace Raven.Server.Documents.Indexes.Static.TimeSeries
+{
+    public class DynamicTimeSeriesSegment : AbstractDynamicObject
+    {
+        private TimeSeriesItem _item;
+
+        public override dynamic GetId()
+        {
+            if (_item == null)
+                return DynamicNullObject.Null;
+
+            Debug.Assert(_item.Key != null, "_item.Key != null");
+
+            return _item.Key;
+        }
+
+        public override void Set(object item)
+        {
+            _item = (TimeSeriesItem)item;
+        }
+
+        protected override bool TryGetByName(string name, out object result)
+        {
+            Debug.Assert(_item != null, "Item cannot be null");
+
+            if (string.Equals("Entries", name))
```

Comment:
Can actually make this into properties, instead of handling dynamically

Reply:
done